src/lib.rs

```diff
@@ -699,13 +710,74 @@ impl LineEvent {
     }
 }
 
-/// Blocking iterator over events for a given line
+/// Handle for retrieving events from the kernel for a line
+///
+/// In order for userspace to retrieve incoming events on a GPIO,
+/// an event handle must be requested from the chip using
+/// [`Line::events`].
+/// On success, the kernel creates an anonymous file descriptor
+/// for reading events. This structure is the go-between for callers
+/// and that file descriptor.
+///
+/// [`Line::events`]: struct.Line.html#method.events
 #[derive(Debug)]
-pub struct LineEventIterator {
+pub struct LineEventHandle {
+    line: Line,
+    handle_flags: LineRequestFlags,
+    event_flags: EventRequestFlags,
     file: File,
 }
 
-impl Iterator for LineEventIterator {
+impl LineEventHandle {
+    /// Retrieve the next event from the kernel for this line
+    ///
+    /// This blocks while there is not another event available from the
+    /// kernel for the line which matches the subscription criteria
+    /// specified in the `event_flags` when the handle was created.
+    pub fn get_event(&self) -> Result<LineEvent> {
+        let mut data: ffi::gpioevent_data = unsafe { mem::zeroed() };
+        let mut data_as_buf = unsafe {
+            slice::from_raw_parts_mut(
+                &mut data as *mut ffi::gpioevent_data as *mut u8,
+                mem::size_of::<ffi::gpioevent_data>(),
+            )
+        };
+        let bytes_read = nix::unistd::read(self.file.as_raw_fd(), &mut data_as_buf)?;
+
+        if bytes_read != mem::size_of::<ffi::gpioevent_data>() {
+            let e = nix::Error::Sys(nix::errno::Errno::EIO);
+            Err(e.into())
+        } else {
+            Ok(LineEvent(data))
+        }
+    }
+
+    /// Request the current state of this Line from the kernel
+    ///
+    /// This value should be 0 or 1 which a "1" representing that
+    /// the line is active.  Usually this means that the line is
+    /// at logic-level high but it could mean the opposite if the
+    /// line has been marked as being ACTIVE_LOW.
+    pub fn get_value(&self) -> Result<u8> {
+        let mut data: ffi::gpiohandle_data = unsafe { mem::zeroed() };
+        let _ = unsafe { ffi::gpiohandle_get_line_values_ioctl(self.file.as_raw_fd(), &mut data)? };
+        Ok(data.values[0])
+    }
+
+    /// Get the Line information associated with this handle.
+    pub fn line(&self) -> &Line {
+        &self.line
+    }
+}
+
+impl AsRawFd for LineEventHandle {
+    /// Gets the raw file descriptor for the LineEventHandle.
+    fn as_raw_fd(&self) -> RawFd {
+        self.file.as_raw_fd()
+    }
+}
+
+impl Iterator for LineEventHandle {
```

Comment:
I believe the `Iterator` implementation should be able to just use `get_event` directly, correct?  I.e.

```rust
impl Iterator for LineEventHandle {
    fn next(&mut self) -> Option<Result<LineEvent>> {
        // EIO might be something we want to revisit in get_event since this we
        // can't distinguish a real EIO and an empty read EIO
        Some(self.get_event())
    }
}
```